File: stephen/discuss/lib/discuss_web/channels/user_socket.ex
defmodule DiscussWeb.UserSocket do
  use Phoenix.Socket

  channel "comments:*", DiscussWeb.CommentsChannel
  # get "/comments/:id", CommentController, :join, :handle_in

  transport, :websocket, Phoenix.Transports.Websocket

  @impl true
  def connect(_params, socket, _connect_info) do
    {:ok, socket}
  end

  @impl true
  def id(_socket), do: nil
end

File: stephen/discuss/assets/js/user_socket.js

import {Socket} from "phoenix"

let socket = new Socket("/socket", {params: {token: window.userToken}})

socket.connect()

let channel = socket.channel("comments:lobby", {})
channel.join()
  .receive("ok", resp => { console.log("Joined successfully", resp) })
  .receive("error", resp => { console.log("Unable to join", resp) })

document.querySelector('button').addEventListener('click', function() {
  channel.push('comments:hello', {hi: 'there'})
})

export default socket
